### PR TITLE
WIP: Fix py3 problems in custom processors

### PR DESCRIPTION
**Pull request by MichalMMac**
Multiple fixes to get recipes working with `autopkg/py2-to-3` code.

- TextMate2 recipe udpated not to use `TextMateURLProvider`. I don't think there is need for it anymore. Incorporated some changes from #311  93a6f2e and c7280fd
- Fixed urllib and bytes/str problems in custom Processors e4bb7bd, c3d04ea, 998d228, 1bc7b6d, 0567ddb 6c85c0d
- Updated `BarebonesURLProvider` sort function for Python3 3e87777

I kept/added urllib "conditional" imports for python2. Should we drop Python2 imports since the code is not incompatible with it (f-Strings, etc.)?

### CODE REVIEW

File: Praat/PraatVersionFixer.py [hunk 1]
```diff
@@ -44,7 +44,11 @@ class PraatVersionFixer(Processor):
     def read_bundle_info(self, path):
         """Read Contents/Info.plist inside a bundle."""
         plistpath = os.path.join(path, "Contents", "Info.plist")
-        info, _, error = NSPropertyListSerialization.propertyListFromData_mutabilityOption_format_errorDescription_(  # noqa
+        (
+            info,
+            _,
+            error,
+        ) = NSPropertyListSerialization.propertyListFromData_mutabilityOption_format_errorDescription_(  # noqa
```
Comment (nmcspadden) [inline Praat/PraatVersionFixer.py]: Naw. We should murder this. There's absolutely no reason not to just use plistlib.

File: AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py [hunk 2]
```diff
@@ -131,7 +127,7 @@ def get_manifest_data(self, manifest_plist_url):
         """Get manifest(plist) data from a url"""
         manifest_plist_response = self.get_url_response(manifest_plist_url)
         try:
-            manifest_data = readPlistFromString(manifest_plist_response)
+            manifest_data = readPlistFromString(manifest_plist_response.encode())
```
Comment (nmcspadden) [inline AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py]: Does this work for both plistlib.readPlistFromString (Py2) _and_ readPlistFromBytes (Py3)?

File: TextMate/TextMate2.download.recipe [hunk 3]
```diff
@@ -13,29 +13,28 @@ preferences panel.
     <string>com.github.autopkg.download.TextMate2</string>
     <key>Input</key>
     <dict>
+        <key>BASE_URL</key>
```
Comment (nmcspadden) [inline TextMate/TextMate2.download.recipe]: This should be its own PR and we should merge this change into master, and then rebase our py2-to-3 branch on it.

File: TextMate/TextMate2.munki.recipe [hunk 4]
```diff
@@ -23,6 +23,8 @@ preferences panel.
         <string>TextMate2</string>
         <key>BRANCH</key>
         <string>release</string>
+        <key>MACOS</key>
```
Comment (nmcspadden) [inline TextMate/TextMate2.munki.recipe]: Same here - the TextMate changes should be in a separate PR on master.

File: TextMate/TextMate2.download.recipe (hunk 3 above)
Comment (MichalMMac) [reply TextMate/TextMate2.download.recipe]: Done. TextMate changes now in separate PR #318

File: TextMate/TextMate2.munki.recipe (hunk 4 above)
Comment (MichalMMac) [reply TextMate/TextMate2.munki.recipe]: Done. TextMate changes now in separate PR #318

File: AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py (hunk 2 above)
Comment (MichalMMac) [reply AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py]: Good catch. I'll try to use `readPlistFromBytes` for both.

**Comment by nmcspadden** [comment]
I updated this from master so I can experiment with it more easily.

File: AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py (hunk 2 above)
Comment (MichalMMac) [reply AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py]: `readPlistFromBytes` not present in python2 plistlib. `readPlistFromString(manifest_plist_response.encode())` seems to work in the python2 code. However I am testing this on master + this applied change only since `py2-to-3` containes python2 breaking changes. Could you remove these changes (f-strings for instance) since you intend to keep python2 comatibility?

Comment (MichalMMac) [reply AdobeAcrobatPro/AdobeAcrobatProUpdateInfoProvider.py]: `encode()` doesn't appear to have any effect on string object in Python 2.

**Comment by nmcspadden** [comment]
I'm merging this in because I've fixed all the issues, and I want to simplify the dev environment.
